src/features/comments.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment, CommentData } from '../types/Comment';
+import * as commentsApi from '../api/comments';
+
+type CommentsState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+  submitting: boolean;
+};
+
+const initialState: CommentsState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+  submitting: false,
+};
+
+export const initComments = createAsyncThunk(
+  'comments/fetch',
+  (postId: number) => commentsApi.getPostComments(postId),
+);
+
+export const addComment = createAsyncThunk(
+  'comments/add',
+  (data: CommentData) => commentsApi.createComment(data),
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    deleteComment: (state, action: PayloadAction<number>) => {
```

Comment:
let's create an async thunk for this async action too
at least you may change local state on `fullfiled` to avoid unnecessary state change if smth went wrong and comment weren't deleted on the server

Reply:
Для проходження тестів потрібно щоб коментар видалявся з інтерфейсу одразу, якщо зробити як ви запропонували і видаляти коментар тільки після видалення на сервері, а якщо станеться помилка видалення на сервері то не видаляти зі стейту, то тоді не проходитимуть тести. То ж видалення залишив одразу, а в async thunk переніс тільки видалення на сервері і додав вивід помилки при неможливості видалення.